lib/get-babel-options.js

```diff
@@ -0,0 +1,79 @@
+const babelOptionsUtil = require('./babel-options-util');
+
+module.exports = function getBabelOptions(config, appInstance) {
+  let { parent, project } = appInstance;
+  let addonProvidedConfig = babelOptionsUtil._getAddonProvidedConfig(config);
+  let shouldCompileModules = babelOptionsUtil._shouldCompileModules(config, project);
+  let shouldIncludeHelpers = babelOptionsUtil._shouldIncludeHelpers(config, appInstance);
+  let shouldHandleTypeScript = babelOptionsUtil._shouldHandleTypeScript(config, parent);
+  let shouldIncludeDecoratorPlugins = babelOptionsUtil._shouldIncludeDecoratorPlugins(config);
```

Comment:
I'd like to destructure these up above, so we are just calling functions directly (basically the same comment as my other one RE: relying on `this`)